docs/learning/WriteAConnector.md

```diff
@@ -1,108 +1,727 @@
+
+For each iTwin Connector author, there will always be two conflicting goals:
 
-* [Open a local briefcase copy](./backend/IModelDb.md) of the iModel that is to be updated.
-* Import or Update Schema
-  * Possibly [import an appropriate BIS schema into the briefcase](./backend/SchemasAndElementsInTypeScript.md#importing-the-schema)  or upgrade an existing schema.
-  * [Push](./backend/IModelDbReadwrite.md#pushing-changes-to-imodelhub) the results to the iModelServer.
-* Convert Changed Data
-  * Connect to the data source.
-  * Detect changes to the source data.
-  * [Transform](../learning/imodel-connectors.md#data-alignment) the new or changed source data into the target BIS schema.
-  * Write the resulting BIS data to the local briefcase.
-  * Remove BIS data corresponding to deleted source data.
-  * Obtain required [Locks and Codes](./backend/ConcurrencyControl.md) from the iModel server and/or code server.
-  * [Push](./backend/IModelDbReadwrite.md#pushing-changes-to-imodelhub) changes to the iModel server.
+1. To transform the data in such a way that it appears logical and "correct" to the users of the authoring application.
+2. To transform the data in such a way that data from disparate authoring applications appear consistent.
 
-## Writing a connector
+The appropriate balancing of these two conflicting goals is not an easy task. However, where clear BIS schema types exist, they should always be used.
 
-A connector would import the following packages:
+#### Schemas
 
-``` ts
-[[include:Bridge.imports.example-code]]
+See this article on [Importing a schema and bootstrapping definitions](https://github.com/imodeljs/imodeljs/tree/master/docs/learning/backend/SchemasAndElementsInTypeScript.md#importing-the-schema)
+
+There are roughly three degrees of customizations you may need to employ to connect and align your data to an iModel. These degrees range from no customization at all (i.e., using the out-of-the-box domain schemas used by many of the Bentley authored Connectors) to extending the domain schemas to introduce additional classes (or subclasses) and properties. Finally, the most extreme level of customization which is to add classes and properties programmatically as your data is read (a.k.a, "dynamic schema").
+
+#### Domain Schemas
+
+Bentley has authored many "domain" schemas to support connectors for many of its authoring applications. For the most aligned data (i.e., data published from your Connector uses the same classes and properties as data published from other connectors), it is best to use a domain schema.
+
+To see what domain schemas are available, see [Released Domain Schemas](ReleasedDomainSchemas.md)
+
+Sometimes BIS domain schemas are not adequate to capture all the data in the authoring application. The flow chart below can be used to assist in deciding which schema methodology to use.
+
+![Schema Methodology Decision](./SchemaDecision.PNG)
+
+#### Extending a Schema

```

Comment:
Did you intend for this heading to be the same level as Dynamic Schemas and Display Labels?

Reply:
Changed header to be on same level as schemas